app/code/Magento/Bundle/Model/LinkManagement.php

```diff
@@ -418,4 +421,17 @@ private function getOptions(ProductInterface $product)
 
         return $optionCollection->appendSelections($selectionCollection, true);
     }
+
+    /**
+     * Get MetadataPool instance
+     *
+     * @return MetadataPool
+     */
+    private function getMetadataPool()
```

Comment:
Why you need this method? Metadatapool os already injected.

Reply:
@sidolov I didn't add this. It looks like this file was reworked after I opened this pull request, including the removal of this function. In commit [e535d373ddad664d7484acf841f94f50fd85c8fa](https://github.com/magento/magento2/commit/e535d373ddad664d7484acf841f94f50fd85c8fa#diff-9af9e82493d309f07e54230e0b313a71). I'll cleanup this file to be up-to-date with this rework.